oceannavigator/frontend/src/components/OceanNavigator.jsx

```diff
@@ -123,6 +123,16 @@ export default class OceanNavigator extends React.Component {
     this.updateScale = this.updateScale.bind(this);
     this.get_variables_promise = this.get_variables_promise.bind(this);
     this.get_timestamp_promise = this.get_timestamp_promise.bind(this);
+    this.setStartTime = this.setStartTime.bind(this);
+    // Setting the starttime and time variable default value
+    const time_promise = this.get_timestamp_promise("giops_day", "votemper");
+      $.when(time_promise).done(function(time) {
+        let newTime = time[time.length-1].id;
+        let newStarttime = time[0].id;
+        this.state.time = newTime;
+        this.state.starttime = newStarttime;
+        this.setState(state);
+      }.bind(this));
```

Comment:
I'm a bit concerned with firing a promise in the constructor of a React object, since `setState` will cause the component to re-render a bunch of times....

could this `time_promise` be put inside a `componentDidMount()`?

See docs: https://reactjs.org/docs/react-component.html#componentdidmount

Reply:
Moved the `time_promise` promise to the` componentDidMount()` lifecycle hook.